### ISSUE TITLE
NFC: SLIX no longer works

### ISSUE THREAD

**Issue by exussum12** [flipperdevices/flipperzero-firmware#3212]
### Describe the bug.

Both previously saved and trying to read a new card fail on the current firmware. 

Flipper displays emulating iso15694-3 (unknown) when trying to emulate a previously working file 

### Reproduction

See above

**Comment by gornekich** [comment]
Hello @exussum12 !
Could you please provide your file dump? It will help to debug the issue.

**Comment by exussum12** [comment]
Hey I can send it to someone on Discord or via email - whats best to send?

**Comment by gornekich** [comment]
Please ,send me on Discord. My nick is gornekich

**Comment by gornekich** [comment]
Still working on this issue. Planning to include support of all older file versions in next release 0.97

**Comment by tdimov93** [comment]
hey @gornekich, I'm also noticing a couple of issues related to SLIX cards (SLIX2 in my case):

- after reading a tag if I click on _More -> Info_, the Flipper reboots and shows a "NULL pointer dereference" message upon startup
- trying to click on a saved SLIX2 file results in a "Cannot load key file", even if it was created by the flipper itself a second or two ago

I saw this running the official firmware v0.95.0

Please let me know if you'd like me to open a separate issue for this or if you think it might be related to the original issue (or handled as part of it).
I'd be happy to assist with more information if necessary

**Comment by gornekich** [comment]
Hi @tdimov93 
Thanks for information. I will work on SLIX support improvements soon and will ask you to reproduce this issues after I am done. No need in opening another issue, let's discuss it here.

**Comment by tdimov93** [comment]
great, thanks!

Unfortunately I'll only have access to the problematic tag for another 6 days. I have the saved file that refuses to open so I'll be able to reproduce the 2nd issue after that without problems but the first issue would be impossible for me to reproduce after Tuesday. Let me know if I can help with grabbing some logs/traces that might help you identify or reproduce it on your end as I don't imagine you'll get around to working on it this week

**Comment by exussum12** [comment]
I have a bunch of tags I can scan if needed

**Comment by tdimov93** [comment]
Do you have any password-protected ones? I just received a pack of 20 fresh SLIX2 tags and I didn't manage to reproduce either of the 2 issues that I mentioned with them. It would seem the problems arise when the flipper fails to read the password-protected data section of the tag

**Comment by exussum12** [comment]
Yeah I have passworded ones. The old firmware could unlock

**Comment by tdimov93** [comment]
Which version are you referring to when you say old firmware? 0.94.1?
Btw is there a way to grab the commands and responses in order to debug NFC with the flipper zero without an external tool? Haven't been able to find info on that

**Comment by exussum12** [comment]
0.93 I think was the latest I could use.

**Comment by exussum12** [comment]
Just tried 0.97.1 and this is sill broken

**Comment by tdimov93** [comment]
Same. Regarding the issue I described, I compared the corrupt NFC file dump with a regular empty SLIX2 tag dump and saw that the problematic one has an empty `Security Status: ` section. When I copied the section from the empty tag file over manually and saved it (it's a bunch of 0s), the flipper was able to parse it successfully. My guess is that given the original tag is likely password-protected, the flipper fails to read that section and ends up writing an empty data array instead of populating it with question marks for example and leaves the file corrupt.
It would also explain the NULL pointer dereference crash when trying to go into the info section right after reading the tag.
